RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1375 @@
+                }
+                if (gch < f01HEDigis.size()) {
+                  // NOTE: assume that soi is high only for a single guy!
+                  //   which must be the case. cpu version does not check for that
+                  //   if that is not the case, we will see that with cuda mmecheck
+                  auto const soibit = soibit_for_sample<Flavor1>(&(f01HEDigis.data()[gch][0]), sample);
+                  if (soibit == 1)
+                    soiSamples[gch] = sampleWithinWindow;
+                } else if (gch >= nchannelsf015) {
+                  auto const soibit = soibit_for_sample<Flavor3>(&(f3HBDigis.data()[gch - nchannelsf015][0]), sample);
+                  if (soibit == 1)
+                    soiSamples[gch] = sampleWithinWindow;
+                }
+              }
+            }
+            alpaka::syncBlockThreads(acc);
+
+            //Loop over each channel, compute input for multifit
+            for (auto channel : uniform_group_elements_y(acc, group, nchannels)) {
+              auto const gch = channel.global;
+              auto const lch = channel.local;
+
+              //Loop over sample
+              for (auto i_sample : independent_group_elements_x(acc, nsamplesForCompute)) {
+                auto const sampleWithinWindow = i_sample;
+                auto const sample = i_sample + startingSample;
+                auto const linearThPerBlock = i_sample + channel.local * nsamplesForCompute;
+
+                // initialize all output buffers
+                if (sampleWithinWindow == 0) {
+                  outputGPU.did()[gch] = 0;
+                  outputGPU.energyM0()[gch] = 0;
+                  outputGPU.timeM0()[gch] = 0;
+                  outputGPU.energy()[gch] = 0;
+                  outputGPU.chi2()[gch] = 0;
+                }
+
+                // offset output
+                auto* amplitudesForChannel = amplitudes + nsamplesForCompute * gch;
+                auto* noiseTermsForChannel = noiseTerms + nsamplesForCompute * gch;
+                auto* electronicNoiseTermsForChannel = electronicNoiseTerms + nsamplesForCompute * gch;
+
+                // configure shared mem
+                float* shrEnergyM0PerTS = alpaka::getDynSharedMem<float>(acc);
+                float* shrChargeMinusPedestal = shrEnergyM0PerTS + nsamplesForCompute * nchannels_per_block;
+                float* shrMethod0EnergyAccum = shrChargeMinusPedestal + nsamplesForCompute * nchannels_per_block;
+                float* shrEnergyM0TotalAccum = shrMethod0EnergyAccum + nchannels_per_block;
+                unsigned long long int* shrMethod0EnergySamplePair =
+                    reinterpret_cast<unsigned long long int*>(shrEnergyM0TotalAccum + nchannels_per_block);
+                if (sampleWithinWindow == 0) {
+                  shrMethod0EnergyAccum[lch] = 0;
+                  //TODO: check this conversion
+                  shrMethod0EnergySamplePair[lch] = int(std::numeric_limits<float>::min());
```

Comment:
`std::numeric_limits<float>::min()` is `1.17549e-38f`.
`(int) std::numeric_limits<float>::min()` is `0`.

Reply:
Here is the original line from CUDA version
https://github.com/cms-sw/cmssw/blob/1234e950f3ee35b6f39abdeba60b2d1a53c0c891/RecoLocalCalo/HcalRecProducers/src/MahiGPU.cu#L165
so also this line should use `__float_as_uint` / `edm::bit_cast`. Good catch!

Reply:
Thanks for catching it. I've implemented it in dceaf88021ca9c616ba0754cd7a6accfe2b6c9d1